apps/guardian-ui/src/setup/FederationSetup.tsx

```diff
@@ -87,16 +88,38 @@ export const FederationSetup: React.FC = () => {
   let canGoBack = false;
   let canRestart = false;
   let content: React.ReactNode;
+  let roleQueryParam: string | null = null;
 
   switch (progress) {
     case SetupProgress.Start:
       if (needsTosAgreement) {
         title = t('setup.progress.tos.title');
         content = <TermsOfService next={() => setNeedsTosAgreement(false)} />;
       } else {
-        title = t('setup.progress.start.title');
-        subtitle = t('setup.progress.start.subtitle');
-        content = <RoleSelector next={handleNext} />;
+        roleQueryParam = getQueryParam('role');
```

Comment:
Prefer using a hook for this, this isn't re-render safe (e.g. if query param changed, it would not trigger the component to re-render.)

Reply:
Eh, feel free to ignore this. This is more of a consideration when using `react-router` or something of its ilk. Since this app doesn't actually do any routing, it's a bit of a non-issue. Although moving this code to outside of the component may make it more explicitly clear that the query params are not re-render safe, and should not be used to manipulate app state.